train.py

```diff
@@ -87,12 +87,21 @@ def train(
         # TODO: add learning rate scheduling
 
         # evaluate the loss on train/val sets and write checkpoints
-        if iter_num > 0 and iter_num % eval_interval == 0 and fabric.global_rank == 0:
+        if iter_num > 0 and iter_num % eval_interval == 0:
             val_loss = validate(fabric, model, val_data)
             fabric.print(f"step {iter_num}: val loss {val_loss:.4f}")
-            # TODO: Save with Fabric
-            # print(f"saving checkpoint to {out_dir}")
-            # torch.save(checkpoint, os.path.join(out_dir, 'ckpt.pt'))
+
+            # TODO: Update this to `fabric.save()` once it supports FSDP checkpointing
+            from torch.distributed.fsdp import FullyShardedDataParallel as FSDP
```

Comment:
let's maybe condition this on the strategy being FSDP
(I'm experimenting with training and I'm just training a smaller model without FDSP)

Reply:
That's a good idea, done that. Since boilerplate was getting a bit large for the train.py, I moved it to a utils.py file, is that fine?